### ISSUE TITLE
Redis::TimeoutError after verifying nominal latency with redis DB.

### ISSUE THREAD

**Issue by MoHayat** [mperham/sidekiq#5244]
Ruby version: 2.6.9
Rails version: 6.0.3.2
Sidekiq / Pro / Enterprise version(s): 6.1

Please include your initializer, sidekiq.yml, and any error message with the full backtrace.

I plan on using an ec2 instance to spin up my workers service.

after launching my ec2 instance, I install the redis cli and test the latency while attempting to connect to an elasticache cluster in my VPC (name: lens-staging-cache-queue) and see that the latency is around 500us (microseconds), which is nominal.

After verifying the latency through the redis cli, I install git, clone the repository containing the code for my worker service, install all necessary dependencies and spin up the sidekiq server using the command "bundle exec sidekiq  -e <env> -q mailers -q default". Unfortunately, even with a timeout of 10 seconds, I still get redis timeout error issues.

I created an AWS tech support ticket, and we were able to isolate the problem from the EC2 instance or the Elasticache cluster it was trying to connect to, so this seems to be purely an issue with how sidekiq is setup. That being said, this config works well in our current setup in GCP, so I'm not sure where to start looking for a solution. Any help would be greatly appreciated.

initializer:
`require 'sidekiq/web'

if %w[production staging].include? ENV['RAILS_ENV']
  Sidekiq.configure_server do |config|
    config.redis = { url: <removed>}
  end

  Sidekiq.configure_client do |config|
    config.redis = { url: <removed>}
  end
end

# Configure Sidekiq-specific session middleware
Sidekiq::Web.use ActionDispatch::Cookies
Sidekiq::Web.use ActionDispatch::Session::CookieStore

if Rails.env.production? || Rails.env.staging?
  Sidekiq::Web.use Rack::Auth::Basic do |username, password|    
    secure_compare_username = ActiveSupport::SecurityUtils.secure_compare(<removed>, username)
    secure_compare_password = ActiveSupport::SecurityUtils.secure_compare(<removed>, password)
    secure_compare_username && secure_compare_password
  end
end`

sidekiq.yml:
`queues:
  - ["mailers", 2]
  - ["default", 1]
mailers:
  :concurrency: 1
default:
  :concurrency: 1`

If you are using an old version, have you checked the changelogs to see if your issue has been fixed in a later version? yes.

https://github.com/mperham/sidekiq/blob/main/Changes.md
https://github.com/mperham/sidekiq/blob/main/Pro-Changes.md
https://github.com/mperham/sidekiq/blob/main/Ent-Changes.md

**Comment by MoHayat** [comment]
here you go! To clarify, I used '-e staging' in the command to produce this error. <img width="1506" alt="sidekiq-error" src="https://user-images.githubusercontent.com/31362591/158656506-2bc0b201-ce8d-4f20-92e7-b04ebe0d7d71.png">

**Comment by MoHayat** [comment]
Odd, I am able to run the following command in the same ec2 instance and interact with the redis DB that way:

`redis-cli -h redis-url-here --tls -p 6379`

would that still be a firewall issue?

**Comment by mperham** [comment]
I don’t know. I’m not saying that is your problem, just that’s what it
looks like. Are you using the exact URL elasticache gave you? You might
also be using the wrong protocol, redis: vs rediss:, plaintext vs TLS.

**Comment by MoHayat** [comment]
I am using the one of the master node and have made sure that there are no spelling errors in there, but I haven't looked into whether it was plaintext vs. TLS.

I'll look into that and update this thread, appreciate the help!

**Comment by MoHayat** [comment]
Apologies for the silence, thought I'd share a quick update on this for future reference.

After toying around with both instances for some time, I kept running into the same latency issues regardless of config, instance sizes/numbers, or setup. After much research, I decided to replace the ElastiCache cluster with a second EC2 instance running redis-server, which seems to have worked.

This solution will work for quite some time, but I'm still looking into the connection issues since we will need to eventually switch to ElastiCache as our platform grows.
